vslib/src/sai_vs_hostintf.cpp

```diff
@@ -36,10 +37,278 @@ typedef struct _hostif_info_t
+
+        status = vs_generic_get(SAI_OBJECT_TYPE_BRIDGE_PORT, bridge_port_id, 1, &attr);
+
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            break;
+        }
+
+        bridge_id = attr.value.oid;
+
+        SWSS_LOG_DEBUG("found bridge %s for port %s",
+                sai_serialize_object_id(bridge_id).c_str(),
+                sai_serialize_object_id(port_id).c_str());
+
+        attr.id = SAI_BRIDGE_ATTR_TYPE;
+
+        status = vs_generic_get(SAI_OBJECT_TYPE_BRIDGE, bridge_id, 1, &attr);
+
+        if (status != SAI_STATUS_SUCCESS)
+        {
+            break;
+        }
+
+        bridge_type = (sai_fdb_entry_bridge_type_t)attr.value.s32;
+
+        SWSS_LOG_DEBUG("bridge %s type is %d",
+                sai_serialize_object_id(bridge_id).c_str(),
+                bridge_type);
+
+        break;
+    }
+}
+
+void process_packet_for_fdb_event(
+        _In_ const uint8_t *buffer,
+        _In_ size_t size,
+        _In_ const std::shared_ptr<hostif_info_t> &info)
+{
+    MUTEX();
+
+    SWSS_LOG_ENTER();
+
+    uint32_t frametime = (uint32_t)time(NULL);
+
+    /*
+     * We add +2 in case if frame contains 1Q VLAN tag.
+     */
+
+    if (size < (sizeof(ethhdr) + 2))
+    {
+        SWSS_LOG_WARN("ethernet frame is too small: %zu", size);
+        return;
+    }
+
+    const ethhdr *eh = (const ethhdr*)buffer;
+
+    uint16_t proto = htons(eh->h_proto);
+
+    uint16_t vlan_id = DEFAULT_VLAN_NUMBER;
+
+    switch (proto)
+    {
+        case ETH_P_IP:
+        case ETH_P_IPV6:
+        case ETH_P_ARP:
+        case ETH_P_RARP:
+
+            {
+                // IP frame, we need to get vlan if from port
+
+                sai_attribute_t attr;
+
+                attr.id = SAI_PORT_ATTR_PORT_VLAN_ID;
+
+                sai_status_t status = vs_generic_get(SAI_OBJECT_TYPE_PORT, info->portid, 1, &attr);
+
+                if (status != SAI_STATUS_SUCCESS)
+                {
+                    SWSS_LOG_WARN("failed to get port vlan id from port %s",
+                            sai_serialize_object_id(info->portid).c_str());
+                    return;
+                }
+
+                vlan_id = attr.value.u16;
+
+                break;
+            }
+
+        case ETH_P_8021Q:
+
+            {
+                // this is tagged frame, get vlan id from frame
+
+                uint16_t tci = htons(eh->h_proto);
+
+                vlan_id = tci & 0xfff;
+
+                if (vlan_id == 0 || vlan_id == 0xfff)
+                {
+                    SWSS_LOG_WARN("invalid vlan id %u in ethernet frame on %s", vlan_id, info->name.c_str());
+                    return;
+                }
+
+                break;
+            }
+
+        default:
+            {
+                SWSS_LOG_WARN("unknown ethernet protocol: 0x%x on %s", proto, info->name.c_str());
+                return;
+            }
+    }
+
+    // we have vlan and mac addres which is KEY, so just see if that is already defined
```

Comment:
>    // we have vlan and mac addres which is KEY, so just see if that is [](start = 0, length = 71)

if the port is port-based router interface, or lag-based router interface, we should skip mac address.

Reply:
what do you mean skip mac address?

Reply:
skip mac address learning, sorry.

Reply:
what do you mean skip mac ? just skip entire fdb notification processing?

Reply:
this condition is getting complicated, do you mean https://github.com/opencomputeproject/SAI/blob/master/inc/sairouterinterface.h#L101
if this attribute is set to port or lag, that is based on port then we skip learning ?

Reply:
fixed